core/src/main/java/net/opentsdb/query/processor/groupby/GroupByNumericArrayIterator.java

```diff
@@ -347,7 +348,7 @@ public void run() {
             // AGG
             try {
               final TimeSeries source = sources != null ? sources.get(startIndex) :
-                result.downstreamResult().timeSeries().get(startIndex);
+                result.downstreamResult().timeSeries().get(arraySources[startIndex]);
```

Comment:
Is it possible for arraySources to be null ? (I see one of the constructors setting it to null).

Reply:
Yeah, that's why we have the `sources != null` check.